pkg/config/config.go

```diff
@@ -1777,10 +1777,27 @@ func ResolveSecrets(config Config, secretResolver secrets.Component, origin stri
 		err = secretResolver.ResolveWithCallback(
 			yamlConf,
 			origin,
-			func(yamlPath []string, value any) {
+			func(yamlPath []string, value any) bool {
 				settingName := strings.Join(yamlPath, ".")
+
+				// We received a notification about an unknown setting. This means that the a value
```

Comment:
Stupid question maybe, but would it help if we had a viper method to set a key by []string path, instead of a dotted path? For example extract second-half of `Set` and call it directly: https://github.com/datadog/viper/blob/v1.1.1/viper.go#L1154-L1156

Reply:
It would partially help for sure. But the configuration allowing slice, maps and struct to be used might still produce some corner cases (especially with `IsSet` and `IsKnown` method from viper).